confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/DeleteTimeSeriesProcedure.java

```diff
@@ -0,0 +1,611 @@
+        currentFailedDataNodeMap.forEach(dataNodeConsensusGroupIdMap::remove);
+        // remove dataNodes that successfully executed request
+        allFailedDataNodeSet.removeAll(dataNodeConsensusGroupIdMap.keySet());
+        dataNodeConsensusGroupIdMap =
+            getAvailableDataNodeLocationForRetry(currentFailedDataNodeMap, allFailedDataNodeSet);
+        // some consensus group has no available dataNode
+        if (isFailed()) {
+          return;
+        }
+      }
+    }
+
+    private Map<TDataNodeLocation, List<TConsensusGroupId>> sendSchemaRegionRequest(
+        Map<TDataNodeLocation, List<TConsensusGroupId>> dataNodeConsensusGroupIdMap) {
+      // send request to each dataNode
+      Map<TDataNodeLocation, List<TConsensusGroupId>> failedDataNodeMap = new HashMap<>();
+      for (Map.Entry<TDataNodeLocation, List<TConsensusGroupId>> entry :
+          dataNodeConsensusGroupIdMap.entrySet()) {
+        // process response
+        Map<Integer, TSStatus> dataNodeResponseMap = sendRequest(entry.getKey(), entry.getValue());
+        TSStatus currentDataNodeResponse = dataNodeResponseMap.get(entry.getKey().getDataNodeId());
+        if (currentDataNodeResponse.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+          if (currentDataNodeResponse.getCode() == TSStatusCode.MULTIPLE_ERROR.getStatusCode()) {
+            // dataNode execution error
+            LOGGER.error(
+                "Failed to execute [{}] of delete timeseries {} on {}",
+                taskName,
+                requestMessage,
+                entry.getKey());
+            setFailure(
+                new ProcedureException(
+                    String.format(
+                        "Delete timeseries %s failed on when [%s]", requestMessage, taskName)));
+            break;
+          } else {
+            // unexpected error, retry on other replicates on other dataNodes
+            failedDataNodeMap.put(entry.getKey(), entry.getValue());
+          }
+        }
+      }
+      return failedDataNodeMap;
+    }
+
+    private Map<TDataNodeLocation, List<TConsensusGroupId>> getAvailableDataNodeLocationForRetry(
+        Map<TDataNodeLocation, List<TConsensusGroupId>> failedDataNodeConsensusGroupIdMap,
+        Set<TDataNodeLocation> allFailedDataNodeSet) {
+      Map<TConsensusGroupId, Integer> leaderMap =
+          env.getConfigManager().getPartitionManager().getAllLeadership();
+      Map<TDataNodeLocation, List<TConsensusGroupId>> availableDataNodeLocation = new HashMap<>();
+      for (List<TConsensusGroupId> consensusGroupIdList :
+          failedDataNodeConsensusGroupIdMap.values()) {
+        for (TConsensusGroupId consensusGroupId : consensusGroupIdList) {
+          TRegionReplicaSet regionReplicaSet = targetSchemaRegionGroup.get(consensusGroupId);
+          TDataNodeLocation selectedDataNode = null;
+          Integer leaderId = leaderMap.get(consensusGroupId);
+          if (leaderId == null || leaderId == -1) {
```

Comment:
Better add annotation to response to the next annotation "retry on the new leader as possible".

Reply:
Added.